packages/phishing-controller/src/utils.ts

```diff
@@ -15,39 +17,52 @@ export const fetchTimeNow = (): number => Math.round(Date.now() / 1000);
  *
  * @param listState - the stalelist or the existing liststate that diffs will be applied to.
  * @param hotlistDiffs - the diffs to apply to the listState if valid.
+ * @param listKey - the key associated with the input/output phishing list state.
  * @returns the new list state
  */
 export const applyDiffs = (
-  listState: PhishingStalelist,
+  listState: PhishingListState,
   hotlistDiffs: Hotlist,
+  listKey: ListKeys,
 ): PhishingListState => {
+  // filter to remove diffs that were added before the lastUpdate time.
+  // filter to remove diffs that aren't applicable to the specified list (by listKey).
   const diffsToApply = hotlistDiffs.filter(
-    ({ timestamp }) => timestamp > listState.lastUpdated,
+    ({ timestamp, targetList }) =>
+      timestamp > listState.lastUpdated &&
+      (targetList.split('.')[0] as ListKeys) === listKey,
```

Comment:
Nit: It'd be nice to have a unit test that demonstrates that this condition works (e.g. a diff for a different list key gets ignored). Maybe this is already covered, but I wasn't able to find a test that demonstrates this.

Reply:
Added a test for this.